src/views/game_view.cpp

```diff
@@ -154,7 +154,38 @@ void GameView::rasterize()
   }
 }
 
+void flip_callback(void *gameview){
+  GameView *gv = (GameView*)gameview;
+  gv->flip();
+}
+
+void GameView::flip()
+{
+  // Start with black
+  manager->clear(0, 0, 0);
 
+  // Create pixels to display from machine memory
+  rasterize();
+  _output.update();
+
+  // Write it to the SDL Surface
+  SDL_Rect dest;
+  if (_scaler == Scaler::UNSCALED)
+    dest = { (SCREEN_WIDTH - 128) / 2, (SCREEN_HEIGHT - 128) / 2, 128, 128 };
+  else if (_scaler == Scaler::SCALED_ASPECT_2x)
+    dest = { (SCREEN_WIDTH - 256) / 2, (SCREEN_HEIGHT - 256) / 2, 256, 256 };
+  else
+    dest = { 0, 0, SCREEN_WIDTH, SCREEN_HEIGHT };
+  manager->blitToScreen(_output, dest);
+
+  // We aren't returning control to ::loop, so force this to display now
+  manager->flip();
+
+  // Wait out the right time between frames
+  // (assumes zero draw time, so this can be improved)
+  int32_t fps = machine.code().require60fps() ? 60 : 30;
+  SDL_Delay(1000.0/fps);
+}

```

Comment:
Libretto cores should use no delay. Instead they are synced in frontend on produced frames and audio samples

Reply:
Yes, in general the `flip()` function is a poor design choice because the running loop should be controlled by the caller instead that letting the code itself doing quirky things.

I guess the solution would be a quick return and a synchronization through a variable on backend which are not synched and just wait for the next frame on others?